include/libpmemobj++/experimental/mpsc_queue.hpp

```diff
@@ -43,22 +47,57 @@ class mpsc_queue {
 		size_t len;
 
 		struct iterator {
-			iterator(char *data) : data(data)
+			iterator(char *data, char *end) : data(data), end(end)
 			{
+				/* Advance to first, unconsumed element */
+				while (this->data < end &&
+				       reinterpret_cast<first_block *>(
+					       this->data)
+						       ->size == 0) {
```

Comment:
so, the `size` field indicates whether the element was consumed or not? (but this makes no sense considering the implementation below)
This needs documentation.